app/controllers/admin/tours_controller.rb

```diff
@@ -42,7 +77,31 @@ def set_tour
     flash[:danger] = t "flash.tour.not_exist"
   end
 
+  def set_uploaded_images
+    @uploaded_images = filter_uploaded_images(params.dig(:tour, :images))
+  end
+
   def tour_params
     params.require(:tour).permit Tour::TOUR_ATTRIBUTES
   end
+
+  def resize_before_save
+    return unless @uploaded_images
+
+    @uploaded_images.each do |image|
+      resize_image(image, Settings.width_img_250, Settings.heigt_img_250)
+    end
+  end
+
+  def resize_image image_param, width, height
+    return unless image_param
+
+    begin
+      image = MiniMagick::Image.read(image_param)
+      image.resize "#{width}x#{height}"
+      image.write(image_param.tempfile.path)
+    rescue StandardError => _e
+      # it's means the => file type is incorrect => model validations.
```

Comment:
Xử lý rescue đi em, in lỗi hoặc điều hướng gì đấy đi chứ

Reply:
Dạ a, e note lại mà quên hiện thông báo